### PR TITLE
Add Auth PKCE to sample app

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add Auth PKCE to sample app.

#### HOW

Add [secrets-gradle-plugin](https://developers.google.com/maps/documentation/android-sdk/secrets-gradle-plugin) to `sample` app and configure secrets to be placed in `secrets.properties` file locally.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: auth-data/src/main/java/com/google/android/horologist/auth/data/pkce/impl/google/AuthPKCEConfigRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+        return AuthPKCEDefaultConfig(
+            clientId = clientId,
+            clientSecret = clientSecret,
+            authProviderUrl = uri
+        )
+    }
+
+    public companion object {
+        public const val ENCODED_PATH: String = "https://accounts.google.com/o/oauth2/v2/auth"
```
Comment (yschimke) [inline auth-data/src/main/java/com/google/android/horologist/auth/data/pkce/impl/google/AuthPKCEConfigRepositoryImpl.kt:50]: Should these constants be in AuthPKCEConfigRepositoryImpl, or is there somewhere else to put them, like a KnownProviders.Google?

File: sample/src/main/java/com/google/android/horologist/auth/oauth/pkce/AuthPKCEScreenViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class AuthPKCEScreenViewModel(application: Application) : AuthPKCEDefaultViewModel(
+    authPKCEConfigRepository = AuthPKCEConfigRepositoryImpl(
+        BuildConfig.AUTH_CLIENT_ID,
+        BuildConfig.AUTH_CLIENT_SECRET
+    ),
+    authPKCEOAuthCodeRepository = AuthPKCEOAuthCodeRepositoryImpl(application),
+    authPKCETokenRepository = AuthPKCETokenRepositoryGoogleImpl(
+        GoogleOAuthServiceFactory(Moshi.Builder().build()).get()
+    ),
```
[the remark this answers is not in the document]
Comment (yschimke) [reply sample/src/main/java/com/google/android/horologist/auth/oauth/pkce/AuthPKCEScreenViewModel.kt:36]: Should we have hilt for the sample app? I guess my worry is that we end up with too complicated real world sample apps?  If we need hilt, could we apply these to the media app? Or leave these samples simpler.

I'm not sure what the right option, just wanted to discuss.

**Review by yschimke — APPROVED** [review]

File: sample/src/main/java/com/google/android/horologist/auth/oauth/pkce/AuthPKCEScreenViewModel.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (luizgrp) [reply sample/src/main/java/com/google/android/horologist/auth/oauth/pkce/AuthPKCEScreenViewModel.kt:36]: No worries, I had impression it was discussed and agreed to go with hilt based on this [comment](https://github.com/google/horologist/blob/8b902f581a99a784a15b701039f1a8e876c605de/sample/src/main/java/com/google/android/horologist/sample/di/SampleAppDI.kt#L43). I share the same concern about the samples being too complicated, happy to work with manual DI in the sample app and hilt in the media sample as suggested 👍

File: auth-data/src/main/java/com/google/android/horologist/auth/data/pkce/impl/google/AuthPKCEConfigRepositoryImpl.kt (hunk 1 above)
Comment (luizgrp) [reply auth-data/src/main/java/com/google/android/horologist/auth/data/pkce/impl/google/AuthPKCEConfigRepositoryImpl.kt]: This file is meant to be a specific to Google implementation, see the package name. I just changed the name to make it more explicit and to match the other file name in this package. 
Not sure if there is value in making it more generic as it is too simple. If you have strong opinions I'm happy to improve it in another PR.

Comment (yschimke) [reply auth-data/src/main/java/com/google/android/horologist/auth/data/pkce/impl/google/AuthPKCEConfigRepositoryImpl.kt]: Makes sense.

File: sample/src/main/java/com/google/android/horologist/auth/oauth/pkce/AuthPKCEScreenViewModel.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (yschimke) [reply sample/src/main/java/com/google/android/horologist/auth/oauth/pkce/AuthPKCEScreenViewModel.kt:36]: I'm ok with Hilt, but I scared developers not exposed to Hilt might be let confused. But use your judgement if you think it's overall simplifying.

[the remark this answers is not in the document]
Comment (luizgrp) [reply sample/src/main/java/com/google/android/horologist/auth/oauth/pkce/AuthPKCEScreenViewModel.kt:36]: https://github.com/google/horologist/pull/763

**Pull request merged**